docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,88 @@
+# Charm architecture
+
+At its core, [Jenkins agents](https://www.jenkins.io/doc/book/managing/nodes/#components-of-distributed-builds) are [Java](https://www.java.com/en/) applications executing the jobs on behalf of [Jenkins](https://www.jenkins.io/) itself.
+
+The charm design leverages the [sidecar](https://kubernetes.io/blog/2015/06/the-distributed-system-toolkit-patterns/#example-1-sidecar-containers) pattern to allow multiple containers in each pod with [Pebble](https://juju.is/docs/sdk/pebble) running as the workload container’s entrypoint.
+
+Pebble is a lightweight, API-driven process supervisor that is responsible for configuring processes to run in a container and controlling those processes throughout the workload lifecycle.
+
+Pebble `services` are configured through [layers](https://github.com/canonical/pebble#layer-specification), and the following containers represent each one a layer forming the effective Pebble configuration, or `plan`:
+
+1. An [Jenkins agent](https://www.jenkins.io/doc/book/managing/nodes/#components-of-distributed-builds) container, which manages the CI jobs.
+
+
+As a result, if you run a `kubectl get pods` on a namespace named for the Juju model you've deployed the Jenkins agent k8s charm into, you'll see something like the following:
+
+```bash
+NAME                             READY   STATUS            RESTARTS   AGE
+jenkins-agent-k8s-0              2/2     Running           0          2m2s
+```
+
+This shows there are 2 containers - the Jenkins agent one and container for the charm code itself.
+
+And if you run `kubectl describe pod jenkins-agent-k8s-0`, all the containers will have as Command ```/charm/bin/pebble```. That's because Pebble is responsible for the processes startup as explained above.
```

Comment:
```suggestion
If you run `kubectl describe pod jenkins-agent-k8s-0`, all the containers will `/charm/bin/pebble` as the entrypoint command. That's because Pebble is responsible for the processes startup as explained above.
```